packages/graphql/src/schema/create-connection-fields.ts

```diff
@@ -125,7 +125,13 @@ function createConnectionFields({
             const relatedNodes = nodes.filter((n) => connectionField.relationship.union?.nodes?.includes(n.name));
 
             relatedNodes.forEach((n) => {
-                const unionWhereName = `${connectionField.typeMeta.name}${n.name}Where`;
+                const connectionName = connectionField.typeMeta.name;
+
+                // Append union member name before "ConnectionWhere"
+                const unionWhereName = `${connectionName.substring(0, connectionName.length - "Connection".length)}${
```

Comment:
**question**: I assume you already checked this and the answer is now, but there is no way of composing this name in a way that does not rely on the use of substring?
It feels a bit flaky

Reply:
I would like to revisit this in the future - pinky promise that the `substring` won't live forever? 🥺